test/lib.bash

```diff
@@ -198,14 +195,9 @@ function run_knative_serving_rolling_upgrade_tests {
       fi
       # Check we got RequirementsNotMet error
       [[ $(oc get ClusterServiceVersion $upgrade_to -n $OPERATORS_NAMESPACE -o=jsonpath="{.status.requirementStatus[?(@.name==\"$upgrade_to\")].message}") =~ "requirement not met: minKubeVersion" ]] || return 1
-      # Check KnativeServing still has the old version
-      [[ $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") == "$serving_version" ]] || return 1
     else
       approve_csv "$upgrade_to" || return 1
-      timeout 900 '[[ ! ( $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") != $serving_version && $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
-
-      # Assert that the old image references eventually fade away
-      timeout 900 "oc get pod -n $SERVING_NAMESPACE -o yaml | grep image: | uniq | grep $serving_version" || return 1
+      timeout 900 '[[ ! ( $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
```

Comment:
I fear that this will return immediately even before the operator upgrades will start. AFAIK, at the beginning it returns True, later it returns False, and then True again when the operators are updated. Not sure how to check that operator images were updated

Reply:
This should pass fine for now as the upgrade is basically a noop. I agree we might need to enhance this a bit to not rely on a version upgrade as much. Shall we do that in a follow up?

Reply:
Regarding the check:  We could poll the openshift-operators namespace for "1.6.0" based images and only go through once they have vanished?

Reply:
Yeah. We can do it in a follow up. The operator images are updated after we approve the installplan, right? In the approve_csv function we wait for the new CSV to be available. Maybe that's enough to check that we have new operators available. But we could also do the check as you suggested.